examples/hooks/components/layout/Tabs.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { cx } from '../../cx';
+
+import './Tabs.css';
+
+export type TabProps = {
+  children: React.ReactNode;
+  title: string;
+};
+
+const getTabKey = (index: number, suffix?: string) =>
+  [`tab-${index}`, suffix].filter(Boolean).join('-');
+
+export function Tabs({ children }) {
+  const [currentTab, setCurrentTab] = useState(0);
+  const tabsRefs = useRef<HTMLElement[]>([]);
+
+  useEffect(() => tabsRefs.current?.[currentTab].focus(), [currentTab]);
+
+  const onKeyDown = (event: React.KeyboardEvent) => {
+    let tabIndex;
+    switch (event.key) {
+      case 'ArrowLeft':
+        tabIndex = Math.max(0, currentTab - 1);
+        break;
+      case 'ArrowRight':
+        tabIndex = Math.min(currentTab + 1, React.Children.count(children) - 1);
+        break;
+      default:
+        return;
+    }
+
+    setCurrentTab(tabIndex);
+  };
```

Comment:
That's quite some code and it couples any "keydown" event with `setCurrentTab`.

How about inlining everything?

```suggestion
  const onKeyDown = (event: React.KeyboardEvent) => {
    if (event.key === 'ArrowLeft') {
      setCurrentTab(Math.max(0, currentTab - 1));
    } else if (event.key === 'ArrowRight') {
      setCurrentTab(
        Math.min(currentTab + 1, React.Children.count(children) - 1)
      );
    }
  };
```